### DISCUSSION TITLE
Is it possible to simply manage NGINX on host via docker deployment?

### DISCUSSION THREAD

**Discussion by Jas-SinghFSU** [0xJacky/nginx-ui#261] [Q&A]
I am running NGINX via Openresty `/usr/local/openresty/nginx` and I want to simply use this application as a UI to modify configurations. Is it possible to use the Docker image to do that? I don't want to deploy another NGINX; strictly only want a nice user interface which this app provides.

**Comment by jtrumpio** [comment]
It should be possible if you host the configs externally (eg on NFS), and then mount the volumes/paths appropriately.  This way ngnix-ui modifies the configs, and your openresty instance reads those configs.  You'll of course have to reload nginx on your openresty install. 

Or in my case, just use nginx-ui itself as a reverse proxy, and set your openresty as the upstream target.

**Comment by Jas-SinghFSU** [comment]
Ah gotcha. Just mount the files so I can edit them but operations like restart will probably need to happen on the host right?

I'm not sure I understand the second part though.

**Comment by 0xJacky** [comment]
Hi @Jas-SinghFSU, if you are running nginx-ui on host, you can modify nginx section of the app.ini file, so that you can reload or restart your nginx docker. Please check our documents for more infomation. https://nginxui.com/guide/config-nginx.html#service-monitoring-and-control

**Comment by jtrumpio** [comment]
> Ah gotcha. Just mount the files so I can edit them but operations like restart will probably need to happen on the host right?
> 
> I'm not sure I understand the second part though.

Here's what I did - essentially nginx acts as a reverse proxy to nginx-ui:

dlb.blah.com (dlb = docker load balancer in my case) is a cname to your host running nginx and the url you'll use to access nginx-ui.  So essentially when you hit dlb.blah.com it will reverse proxy it to the nginx-ui interface.  You can then add your additional site configs.  Depending on your use case you may want to wrap some additional security around this at a minimum ip restrictions for example and of course SSL.  Also as a disclaimer I am not an nginx expert, this is just what I figured out on my own so hopefully it is helpful.


```
upstream dlb {
    server 127.0.0.1:9000;
}
server {
    listen 80;
    listen [::]:80;
    server_name dlb.blah.com;
    return 307 https://$server_name$request_uri;     
    location /.well-known/acme-challenge {
        proxy_set_header Host $host;
        proxy_set_header X-Real_IP $remote_addr;
        proxy_set_header X-Forwarded-For $remote_addr:$remote_port;
        proxy_set_header Upgrade websocket;
        proxy_set_header Connection Upgrade;
        proxy_pass http://127.0.0.1:9180;
    }
}
server {
    listen 443 ssl;
    listen [::]:443 ssl;
    server_name dlb.blah.com;
    ssl_certificate /etc/nginx/ssl/dlb.blah.com/fullchain.cer;
    ssl_certificate_key /etc/nginx/ssl/dlb.blah.com/private.key;
    location /.well-known/acme-challenge {
        proxy_set_header Host $host;
        proxy_set_header X-Real_IP $remote_addr;
        proxy_set_header X-Forwarded-For $remote_addr:$remote_port;
        proxy_pass http://127.0.0.1:9180;
    }
    location / {
        proxy_set_header Upgrade websocket;
        proxy_set_header Connection Upgrade;
        proxy_pass http://dlb;
    }
}
```
